### PR TITLE
Batch of additional tests

### PR DESCRIPTION
**Pull request by rossberg-chromium**
For starters, more tests for:
* block
* unreachable
* nop
* comments

Also introduced a bit of sugar for exports. You can now write an export name as part of the function def itself:
```
(func "func-name" $func-name ...)
```
or, if the symbolic name isn't needed, just
```
(func "func-name" ...)
```
avoiding the tedious triple repetition of the name like before.

### CODE REVIEW

File: ml-proto/test/block.wast [hunk 1]
```diff
@@ -1,35 +1,165 @@
...
+    (block (br 0 (i32.const 8)) (br 0) (br 0 (f64.const 8)))
+    (block (br 0 (i32.const 8)) (br 0 (call $nop)) (br 0 (f64.const 8)))
+    (block (br 0 (i32.const 8)) (br 0) (br 0 (f32.const 8)) (i64.const 3))
+    (block (br 0) (br 0 (i32.const 8)) (br 0 (f64.const 8)) (br 0 (nop)))
+    (block (br 0) (br 0 (i32.const 8)) (br 0 (f32.const 8)) (i64.const 3))
+    (block (block (br 0) (br 1 (i32.const 8))) (br 0 (f32.const 8)) (i64.const 3))
   )
 
-  (func $effects (result i32)
+  (func "effects" $fx (result i32)
```
Comment (kripken) [inline ml-proto/test/block.wast]: why should this be valid? i don't understand what two names for a function means

Comment (binji) [inline ml-proto/test/block.wast]: This is syntactic sugar for:
```
(export "effects" $fx)
(func $fx ...)
```

Comment (kripken) [inline ml-proto/test/block.wast]: Oh, thanks! I missed this part of the issue text...

File: ml-proto/test/comments.wast [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+(;)))))))))));)
+
+(;comment";)
+
+(;comment"";)
+
+(;comment""";)
+
+(;äh,scheiße;)
```
Comment (sunfishcode) [inline ml-proto/test/comments.wast]: Would you mind rewording this to avoid the possibility of appearing unprofessional?

Comment (pjuftring) [inline ml-proto/test/comments.wast]: How about "Heiße Würstchen"? This has umlaut and sharp s as well :)

Comment (rossberg-chromium) [inline ml-proto/test/comments.wast]: Done.

**Comment by sunfishcode** [comment]
lgtm

**Pull request merged**